### PR TITLE
Sibling route presentation mode

### PR DESCRIPTION
**Pull request by Obbut**
This is based on #18, so merge that one first. Check [this diff](https://github.com/owowagency/swiftui-router/pull/19/files/7411e542e530cf49c73dc592dbb5799418fa6f8f..da717c99549100150df60777667e3dda76edaf51) for changes compared to #18.

This PR adds an enum `RoutePresentationMode` that replaces the previous boolean `replacesParent` on `RoutePresenter`.

Before 1.0 it must be discussed if the enum is the right solution, because adding cases will cause a major version bump. I think it is, because the presentation modes must be implemented in the routers, so adding a new one without routers implemented outside of the package being updated will cause undefined behaviour.

I've added this to the list in #4 so it does not have to be blocking to merge this PR.

### CODE REVIEW

File: Sources/Router/Presenter/Presenter.swift [hunk 1]
```diff
@@ -3,11 +3,17 @@ import SwiftUI
 @available(iOS 13, macOS 10.15, *)
 public protocol Presenter {
     associatedtype Body: View
-    var replacesParent: Bool { get }
+    var presentationMode: RoutePresentationMode { get }
     
     func body(with context: PresentationContext) -> Body
 }
 
+public enum RoutePresentationMode {
```
Comment (Joannis) [inline Sources/Router/Presenter/Presenter.swift:11]: Please make this an `internal enum` wrapped by a `struct` with static properties. That allows us to add more presentation modes without breaking API.

File: Sources/Router/Views/RouteViewIdView.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import SwiftUI
+
+@available(iOS 13, macOS 10.15, *)
+struct RouteViewIdView<Content: View>: View, Equatable {
```
Comment (Joannis) [inline Sources/Router/Views/RouteViewIdView.swift]: What's this for?

File: Sources/Router/Presenter/Presenter.swift (hunk 1 above)
Comment (Obbut) [reply Sources/Router/Presenter/Presenter.swift:11]: Please see my comment in the PR description on why I think this is a better solution.

File: Sources/Router/Views/RouteViewIdView.swift (hunk 2 above)
Comment (Obbut) [reply Sources/Router/Views/RouteViewIdView.swift]: It's the reason PR reviews exist 😉 Thanks for noticing, it's a leftover from an earlier attempt. I've removed it.

File: Sources/Router/Presenter/Presenter.swift (hunk 1 above)
Comment (Joannis) [reply Sources/Router/Presenter/Presenter.swift:11]: After discussion, this has been moved to #4

Comment (Obbut) [reply Sources/Router/Presenter/Presenter.swift:11]: Added it more explicitly to #4 after discussion

**Review by Joannis — APPROVED** [review]

**Pull request merged**